pkg/kusionctl/cmd/apply/options.go

```diff
@@ -166,7 +174,15 @@ func preview(o *ApplyOptions, planResources *models.Spec,
 	return operation.NewChanges(project, stack, rsp.Order), nil
 }
 
-func apply(o *ApplyOptions, planResources *models.Spec, changes *operation.Changes) error {
+// The apply function will apply the resources changes,
+// and will save the state to specified storage.
+func Apply(
```

Comment:
Check your pub apis, you'd better not write too many details in the pub api directly, try to keep the core logic/flow clear and readable, and preferably no more than one screen in length.